### PR TITLE
CMake installation rules for 3rd party components 

### PR DESCRIPTION
**Pull request by slyubimt**
Add CMake install rules for OpenVINO 3rd party dependencies
Remove unwanted targets from final distribution (test, static libs, etc.)

### CODE REVIEW

File: inference-engine/cmake/dependencies.cmake [hunk 1]
```diff
@@ -125,6 +125,14 @@ if (THREADING STREQUAL "OMP")
     update_deps_cache(OMP "${OMP}" "Path to OMP root folder")
     log_rpath_from_dir(OMP "${OMP}/lib")
     debug_message(STATUS "intel_omp=" ${OMP})
+    
+    if(WIN32)
+        file(GLOB_RECURSE source_list "${OMP}/*.dll")
+    elseif(LINUX OR APPLE)
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/dependencies.cmake:131]: `LINUX OR APPLE` -> `UNIX`

Comment (ilya-lavrenov) [reply inference-engine/cmake/dependencies.cmake:131]: The proper way is to use https://cmake.org/cmake/help/v3.0/variable/CMAKE_SHARED_LIBRARY_SUFFIX.html to have an unified code here.

File: inference-engine/cmake/dependencies.cmake [hunk 2]
```diff
@@ -176,6 +184,11 @@ if (THREADING STREQUAL "TBB" OR THREADING STREQUAL "TBB_AUTO")
         log_rpath_from_dir(TBB "${TBB}/lib")
     endif ()
     debug_message(STATUS "tbb=" ${TBB})
+    
+    install(DIRECTORY ${TBB}
+        DESTINATION /deployment_tools/inference_engine/external
+        COMPONENT tbb)
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/dependencies.cmake:190]: see https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/inference_engine/CMakeLists.txt#L212-L229

File: inference-engine/cmake/dependencies.cmake [hunk 3]
```diff
@@ -263,6 +276,16 @@ if (ENABLE_OPENCV)
...
     debug_message(STATUS "opencv=" ${OPENCV})
+
+    install(DIRECTORY ${OPENCV}/
+            DESTINATION opencv
+            COMPONENT opencv
+            PATTERN ie_dependency.info EXCLUDE)
+            
+    install(DIRECTORY ${OPENCV}/../python/
+            DESTINATION python
+            COMPONENT opencv)
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/dependencies.cmake:287]: @asuhov @alalek I think we should not install this version of OpenCV. It's used only for tests..

File: scripts/setupvars/setupvars.sh [hunk 4]
```diff
@@ -51,10 +51,10 @@ if [ -e "$INSTALLDIR/deployment_tools/inference_engine" ]; then
 
     export HDDL_INSTALL_DIR=$INSTALLDIR/deployment_tools/inference_engine/external/hddl
     if [[ "$OSTYPE" == "darwin"* ]]; then
-        export DYLD_LIBRARY_PATH=$INSTALLDIR/deployment_tools/inference_engine/external/mkltiny_mac/lib:$INSTALLDIR/deployment_tools/inference_engine/external/tbb/lib:${IE_PLUGINS_PATH}${DYLD_LIBRARY_PATH:+:DYLD_LIBRARY_PATH}
-        export LD_LIBRARY_PATH=$INSTALLDIR/deployment_tools/inference_engine/external/mkltiny_mac/lib:$INSTALLDIR/deployment_tools/inference_engine/external/tbb/lib:${IE_PLUGINS_PATH}${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
+        export DYLD_LIBRARY_PATH=$INSTALLDIR/deployment_tools/inference_engine/external/omp/lib:$INSTALLDIR/deployment_tools/inference_engine/external/mkltiny_mac/lib:$INSTALLDIR/deployment_tools/inference_engine/external/tbb/lib:${IE_PLUGINS_PATH}${DYLD_LIBRARY_PATH:+:DYLD_LIBRARY_PATH}
+        export LD_LIBRARY_PATH=$INSTALLDIR/deployment_tools/inference_engine/external/omp/lib:$INSTALLDIR/deployment_tools/inference_engine/external/mkltiny_mac/lib:$INSTALLDIR/deployment_tools/inference_engine/external/tbb/lib:${IE_PLUGINS_PATH}${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
```
Comment (ilya-lavrenov) [inline scripts/setupvars/setupvars.sh:55]: @AlexPeskov @dmitry-gorokhov do we still need `mkltiny_mac` here?

File: inference-engine/cmake/dependencies.cmake [hunk 5]
```diff
@@ -294,6 +317,16 @@ if (ENABLE_GNA)
     endif()
     update_deps_cache(GNA "${GNA}" "Path to GNA root folder")
     debug_message(STATUS "gna=" ${GNA})
+    
+    if(WIN32)
+        file(GLOB_RECURSE source_list "${GNA}/*.dll")
+        install(FILES ${source_list} DESTINATION ${IE_CPACK_RUNTIME_PATH})
+    elseif(LINUX)
+        file(GLOB_RECURSE source_list "${GNA}/*.so*")
+        install(FILES ${source_list} DESTINATION "deployment_tools/inference_engine/external/gna/lib")
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/dependencies.cmake:326]: see https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/gna_plugin/CMakeLists.txt#L70-L72

File: inference-engine/cmake/dependencies.cmake (hunk 1 above)
Comment (ilya-lavrenov) [reply inference-engine/cmake/dependencies.cmake:131]: BTW, we don't ship packages with OMP.

File: inference-engine/cmake/dependencies.cmake (hunk 3 above)
Comment (asuhov) [reply inference-engine/cmake/dependencies.cmake:287]: Agree, this version is usually a bit outdated. But it's convenient to have it in the package to run samples.

Comment (azhogov) [reply inference-engine/cmake/dependencies.cmake:287]: We should install ALL components that a part of package now, and resolve other issues step by step...
1st priority - no actions out of CMake for preparing FULL package.

File: inference-engine/cmake/dependencies.cmake (hunk 2 above)
Comment (slyubimt) [reply inference-engine/cmake/dependencies.cmake]: removed

File: inference-engine/src/gna_plugin/CMakeLists.txt [hunk 6]
```diff
@@ -67,6 +67,7 @@ set_target_properties(${TARGET_NAME} ${TARGET_NAME}_test_static
 
 # install
 
-install(FILES "${GNA_KERNEL_LIBRARY}"
-        DESTINATION ${IE_CPACK_IE_DIR}/external/gna/lib
+file(GLOB_RECURSE source_list "${libGNA_LIBRARIES_BASE_PATH}/*${CMAKE_SHARED_LIBRARY_SUFFIX}*")
+install(FILES ${source_list} 
+         DESTINATION ${IE_CPACK_IE_DIR}/external/gna/lib
```
Comment (ilya-lavrenov) [inline inference-engine/src/gna_plugin/CMakeLists.txt:72]: @dorloff do we really need to install all libraries?

File: inference-engine/cmake/dependencies.cmake [hunk 7]
```diff
@@ -125,6 +125,10 @@ if (THREADING STREQUAL "OMP")
     update_deps_cache(OMP "${OMP}" "Path to OMP root folder")
     log_rpath_from_dir(OMP "${OMP}/lib")
     debug_message(STATUS "intel_omp=" ${OMP})
+    
+    file(GLOB_RECURSE source_list "${OMP}/*${CMAKE_SHARED_LIBRARY_SUFFIX}*")
+    install(FILES ${source_list} DESTINATION "deployment_tools/inference_engine/external/omp/lib")
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/dependencies.cmake:130]: let's add a component `omp` and it's done for all other installed files.

File: inference-engine/cmake/dependencies.cmake [hunk 8]
```diff
@@ -294,6 +300,7 @@ if (ENABLE_GNA)
     endif()
     update_deps_cache(GNA "${GNA}" "Path to GNA root folder")
     debug_message(STATUS "gna=" ${GNA})
+
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/dependencies.cmake:303]: please, remove empty lines

File: inference-engine/cmake/dependencies.cmake (hunk 7 above)
Comment (slyubimt) [reply inference-engine/cmake/dependencies.cmake]: added

File: inference-engine/cmake/dependencies.cmake [hunk 9]
```diff
@@ -125,6 +125,12 @@ if (THREADING STREQUAL "OMP")
     update_deps_cache(OMP "${OMP}" "Path to OMP root folder")
     log_rpath_from_dir(OMP "${OMP}/lib")
     debug_message(STATUS "intel_omp=" ${OMP})
+    
+    file(GLOB_RECURSE source_list "${OMP}/*${CMAKE_SHARED_LIBRARY_SUFFIX}*")
+    install(FILES ${source_list} 
+    DESTINATION "deployment_tools/inference_engine/external/omp/lib"
+    COMPONENT omp)
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/dependencies.cmake:132]: 1. `ie_cpack_add_component(omp)` first
2. Align next lines of install rules with `(`

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@slyubimt please, merge with latest master to make CI green.

**Comment by ilya-lavrenov** [comment]
@slyubimt please, make CI green

**Comment by slyubimt** [comment]
ci/jenkins green status: https://openvino-ci.intel.com/job/private-ci/job/gitlab_precommit/18232/

**Review by azhogov — APPROVED** [review]

**Comment by azhogov** [comment]
CI passed: https://openvino-ci.intel.com/job/private-ci/job/gitlab_precommit/18232/

**Pull request merged**